src/providers/core/mod.rs

```diff
@@ -142,7 +150,10 @@ impl Provide for Provider {
             // Currently Parsec only stores keys, we delete all of them.
             let keys = provider
                 .list_keys(
-                    ApplicationName::from_name(client.clone()),
+                    ApplicationIdentity {
+                        name: client.clone(),
+                        authenticator_id: AuthType::NoAuth as u8,
+                    }, // TODO: Check auth type
```

Comment:
This is interesting. So we have to delete a client, and this has to happen for the authenticator used by the admin for this request. We have to change the interface for `delete_client`, I assume

Reply:
Yes I would think so. Will work fine currently as the On Disk KIM doesn't rely on the AuthType. Although for the SQLite KIM it will need updating.